### DISCUSSION TITLE
How to Remove Item from the standard toolbar or ribbon control

### DISCUSSION THREAD

**Discussion by abt5555** [DotSpatial/DotSpatial#1512] [Q&A] [answered]
Hello,

I would like to remove some buttons from standard toolbar which comes from SpatialToolStripPanel (I dont use SpatialToolStrip). 

With the code below, I could remove Pan button. 

```
public Form1()
{
            InitializeComponent();
            appManager1.LoadExtensions();
            appManager1.HeaderControl.Remove("Pan");
}
```
but when I add 

`appManager1.HeaderControl.Remove("Zoom In");`

it does not remove Zoom In button.  Is this button name correct? 

My purpose is to get just measure button and scale bar which come with plugins from standard toolbar. I thought I can do it by removing all buttons except measure button and scale bar. And also how to change their size? 

I use VS2022 Net 6.0 Dotspatial 4.0.656

Thank you

**Comment by abt5555** [comment]
I think I found the button names in the default menubar/toolbar. from 

https://github.com/DotSpatial/DotSpatial/blob/master/Source/Core/DotSpatial.Controls/DefaultMenuBars.cs

With these names, I can only remove  Pan and Select buttons where I am doing wrong. Any help?          

```
appManager1.HeaderControl.Remove("Pan");
appManager1.HeaderControl.Remove("Select");
```
If there is a way that I can use measure and webmap plugins in my own created toolbar, it would be great.

Thank you

**Comment by jany-tenaj** [comment, marked as answer]
Have a look at #686 . There is said that you can use AppManager.HeaderControl.RemoveAll() to get rid of all buttons. I don't know if it will work by try whether you can call this before loading the extensions that add the buttons you want.

**Comment by abt5555** [comment]
Thank you so much. It works ! :)

```
        public Form1()
        {
            InitializeComponent();
            appManager1.HeaderControl.RemoveAll();
            appManager1.LoadExtensions();
        }
```
Kind regards,
